File: filters/extract_table_text.lua
-- This script saves the contents of the second column of all tables in a separate markdown file.

function Table (table)
    local column2 = {}
    for i, row in ipairs(table['head']['rows']) do
        -- Extract the content from the cells in the second column.
        for j, cell in ipairs(row) do
            if j == 2 then
                -- The text in the cell is assumed to be a paragraph. 
                -- We extract the text from the first (and presumably only) element.
                table.insert(column2, pandoc.utils.stringify(cell.contents[1]))
            end
        end
    end

    -- Write the contents of the second column to a markdown file.
    local file = io.open("column2.md", "w")
    for i, text in ipairs(column2) do
        file:write("* ", text, "\n")
    end
    file:close()

    return table
end

File: filters/extract_table_text.py
#!/home/jeremy/Python3.10Env/bin/python

import panflute as pf

def action(elem, doc):
    if isinstance(elem, pf.Table):
        # Extract text from the second column
        text_elements = [cell.content[0] for row in elem.content for cell in row.content[2::3]]

        # Create a list of paragraphs from the text elements
        paragraphs = [pf.Para(pf.Str(pf.stringify(te))) for te in text_elements]

        # Replace the table with the paragraphs
        return paragraphs

def main(doc=None):
    return pf.run_filter(action, doc=doc)

if __name__ == '__main__':
    main()
